### PR TITLE
streams: Update documentation (#10330)

### PR DESCRIPTION
**Pull request by jiro4989**
Updated documentation for streams module.

- Basic usage
- See also links
- Runnable examples

My english is not perfect.
Please fix or tell me if english sentence was wrong.

ref #10330

### CODE REVIEW

File: lib/pure/streams.nim [hunk 1]
```diff
@@ -103,6 +271,14 @@ proc readDataStr*(s: Stream, buffer: var string, slice: Slice[int]): int =
 when not defined(js):
   proc readAll*(s: Stream): string =
     ## Reads all available data.
+    ##
+    ## **Note:**
+    ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

File: lib/pure/streams.nim [hunk 2]
```diff
@@ -116,173 +292,447 @@ when not defined(js):
...
+    strm.close()
   read(s, result)
 
 proc peekBool*(s: Stream): bool =
-  ## peeks a bool from the stream `s`. Raises `IOError` if an error occurred.
+  ## Peeks a bool from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("12")
+    doAssert strm.peekBool()
+    doAssert strm.peekBool()
```
Comment (narimiran) [inline lib/pure/streams.nim]: No need to `peek` three times in a row here.

(I get the idea why you did it - you want to contrast it with `read`, but I don't think it is necessary)

File: lib/pure/streams.nim [hunk 3]
```diff
@@ -116,173 +292,447 @@ when not defined(js):
...
+    strm.close()
   read(s, result)
 
 proc peekInt8*(s: Stream): int8 =
-  ## peeks an int8 from the stream `s`. Raises `IOError` if an error occurred.
+  ## Peeks an int8 from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("12")
+    doAssert strm.peekInt8() == 49
+    doAssert strm.peekInt8() == 49
```
Comment (narimiran) [inline lib/pure/streams.nim]: Same as above: no need to `peek` three times in a row here.

File: lib/pure/streams.nim [hunk 4]
```diff
@@ -116,173 +292,447 @@ when not defined(js):
...
+    strm.close()
   read(s, result)
 
 proc peekInt16*(s: Stream): int16 =
-  ## peeks an int16 from the stream `s`. Raises `IOError` if an error occurred.
+  ## Peeks an int16 from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("001020")
+    doAssert strm.peekInt16() == 12336
+    doAssert strm.peekInt16() == 12336
```
Comment (narimiran) [inline lib/pure/streams.nim]: Same as above.

Maybe these examples would work better if you `peek` then `read`, then again `peek` and `read`? Just like you did below for `proc peekLine`.

File: lib/pure/streams.nim [hunk 5]
```diff
@@ -324,25 +813,66 @@ proc readLine*(s: Stream): TaintedString =
...
 
 when not defined(js):
 
   type
-    StringStream* = ref StringStreamObj ## a stream that encapsulates a string
+    StringStream* = ref StringStreamObj
+      ## A stream that encapsulates a string
+      ##
+      ## **Note:**
+      ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

File: lib/pure/streams.nim [hunk 6]
```diff
@@ -324,25 +813,66 @@ proc readLine*(s: Stream): TaintedString =
...
     StringStreamObj* = object of StreamObj
-      data*: string
+      ## A string stream object.
+      ##
+      ## **Note:**
+      ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

File: lib/pure/streams.nim [hunk 7]
```diff
@@ -404,7 +934,24 @@ when not defined(js):
       s.data = nil
 
   proc newStringStream*(s: string = ""): owned StringStream =
-    ## creates a new stream from the string `s`.
+    ## Creates a new stream from the string `s`.
+    ##
+    ## **Note:**
+    ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

File: lib/pure/streams.nim [hunk 8]
```diff
@@ -418,8 +965,16 @@ when not defined(js):
     result.readDataStrImpl = ssReadDataStr
 
   type
-    FileStream* = ref FileStreamObj ## a stream that encapsulates a `File`
+    FileStream* = ref FileStreamObj
+      ## A stream that encapsulates a `File`
+      ##
+      ## **Note:**
+      ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

File: lib/pure/streams.nim [hunk 9]
```diff
@@ -418,8 +965,16 @@ when not defined(js):
...
     FileStreamObj* = object of Stream
+      ## A file stream object.
+      ##
+      ## **Note:**
+      ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

File: lib/pure/streams.nim [hunk 10]
```diff
@@ -447,7 +1002,35 @@ when not defined(js):
       raise newEIO("cannot write to stream")
 
   proc newFileStream*(f: File): owned FileStream =
-    ## creates a new stream from the file `f`.
+    ## Creates a new stream from the file `f`.
+    ##
+    ## **Note:**
+    ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

File: lib/pure/streams.nim [hunk 11]
```diff
@@ -461,17 +1044,67 @@ when not defined(js):
...
+    ## Creates a new stream from the file named `filename` with the mode `mode`.
+    ## If the file cannot be opened, nil is returned. See the `io
+    ## <io.html>`_ module for a list of available FileMode enums.
+    ##
+    ## **Note:**
+    ## * **This function returns nil in case of failure.**
+    ##   To prevent unexpected behavior and ensure proper error handling,
+    ##   use `openFileStream proc <#openFileStream,string,FileMode,int>`_
+    ##   instead.
+    ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## * Not available for JS backend.`

File: lib/pure/streams.nim [hunk 12]
```diff
@@ -461,17 +1044,67 @@ when not defined(js):
...
     var f: File
     if open(f, filename, mode, bufSize): result = newFileStream(f)
 
   proc openFileStream*(filename: string, mode: FileMode = fmRead, bufSize: int = -1): owned FileStream =
-    ## creates a new stream from the file named `filename` with the mode `mode`.
+    ## Creates a new stream from the file named `filename` with the mode `mode`.
     ## If the file cannot be opened, an IO exception is raised.
+    ##
+    ## **Note:**
+    ## * Not available this when backend is js
```
Comment (narimiran) [inline lib/pure/streams.nim]: --> `## **Note:** Not available for JS backend.`

**Comment by jiro4989** [comment]
Thanks for your review. I will fix those.

**Comment by jiro4989** [comment]
@narimiran 
I has fixed your advices.
Please review again.

**Pull request merged**